doc/dependency-resolution.md

```diff
@@ -1,356 +1,129 @@
-            "parent": [
-              {
-                "content_id": "333-333",
-                "title": "Home",
-                "base_path": "/"
-              }
-            ]
-          }
-        ]
-      }
-    ]
-  }
-}
-```
+- As translations. If a content item is available with a different locale it
+  will automatically be linked as a translation of a content item.
+- Content items linking to this item as a parent item. Content items that are
+  linked to as a parent automatically implicitly reciprocate this linking as
+  children of this link.
+- Recursive links. For certain types of links we may want to know not just
+  about the item we have linked to but also the items that item links to and so
+  on. An example of this would be breadcrumbs, where we need only link to our
+  immediate parent item but still want to know about all the links that are
+  parents to that item so we can render a path through the page hierarchy
 
-**Bidirectional Dependencies**
+A key aspect of Dependency Resolution is thus the ability to determine all the
+explicit and implicit links for a Content Item at the time of rendering.
 
-As mentioned, we have a requirement for bidirectional dependencies, in the case
-of the collections requirement. Perhaps the simplest way to satisfy this
-requirement is to manually add a separate group for 'children' that mirrors that
-of the 'parent' group. However, this would place a burden on publishing
-applications to manage this for themselves and it could be problematic if these
-groups fall out of sync.
+The root of this resolution is done using the
+[ExpandedLinkSet presenter][expanded_link_set_presenter]. This collates
+dependees (items which are linked to) with dependents (items that link to this
+content item - eg children) and translations.
 
-Therefore, this proposal recommends that bidirectional dependencies be handled
-explicitly as an additional feature of dependency resolution. I propose we add
-some additional configuration settings in the content schemas. There will be an
-option to state that a particular dependency is the reverse of something else.
-For example, the 'children' group would be marked as the reverse of the 'parent'
-group.
+## Resolving which content items to update
 
-It is important to note that, although these groups are related, they could
-differ in their expansion fields, as well as the recursive nature of their
-expansion. To use the parent / child example, again – a child's parent will form
-part of its breadcrumbs and so its full ancestry would need to be expanded in
-order to render these. A group's children, however, could be presented one level
-at a time and need not be expanded recursively to form the full hierarchy of
-descendents.
+When a content item is updated we need not only update the representation of
+that item in the content store but also the representation of any item that has
+either an implicit or explicit link to a content item.
```

Comment:
It might be good to tweak this slightly, as it could be misleading. In terms of the raw details, you don't actually have to update every content item that you might consider needing updating just from looking at the graph, as it really depends if any of the data that goes in to the expanded links has changed. While at the moment the Publishing API does not do this, it is a possible change that may be made in the future.

I think it could be clearer to always talk about dependency resolution in terms of updating the expanded links of content items (and only talking about the actual links when you get to the implementation details).